trunk-recorder/call.cc

```diff
@@ -179,6 +183,29 @@ void Call::end_call() {
       myfile.close();
     }
 
+    if (sys->get_daily_log()) {
+      std::ofstream myfile2(dailylog_filename, std::ofstream::app);
+      if (myfile2.is_open()) {
+        myfile2 << "\n" << this->get_start_time() << "," << (this->stop_time - this->get_start_time()) << "," << final_length << "," << this->talkgroup << "," << this->emergency << "," << this->priority << "," << this->duplex << "," << this->mode << ",";
+
+        for (int i = 0; i < src_list.size(); i++) {
+          myfile2 << src_list[i].source;
+          if (i < (src_list.size()-1)) {
+            myfile2 << "|";
+          }
+        }
+        myfile2 << this->get_recorder()->get_source()->get_device() << ",";
+
+        for (int i = 0; i < freq_count; i++) {
+          myfile2 << "," << (int)freq_list[i].freq << "|" << freq_list[i].total_len << "|" << freq_list[i].error_count << "|" << freq_list[i].spike_count;
+          //would like to include phase2 slot here
+        }
+        myfile2.close();
+      } else {
+        BOOST_LOG_TRIVIAL(error) << "Can't open daily log file for call: " << this->start_time << "," << (this->stop_time - this->start_time) << "," << (int)(final_length + 0.5) << "," << this->talkgroup << "," << this->emergency << "," << this->priority << "," << this->duplex << "," << this->mode  << "," << sys->get_short_name() << "," << (int)freq_list[0].freq << "|" << freq_list[0].total_len << "|" << freq_list[0].error_count << "|" << freq_list[0].spike_count  ;
```

Comment:
Several questions:

1. Is it necessary to round to the nearest int instead of taking the floor of `final_length`? You do it here, but you don't do it for the log entry being written to the logfile if it was available.

2. You log the system name which makes sense, but can you standardize this to what most of the other loggers in here do, by prefixing the log statement with the system in brackets?

3. What's the purpose of logging all this data as an error if writing to the system daily log file fails? Is it a way to recover data back after the fact? If that's the case, why not also dump the rest of the data that the log file would've gotten?

If that's not the case, can this be less verbose? Stating that the system daily log file isn't openable would be enough for a trunk-recorder instance maintainer to investigate further.

Reply:
1. I didn't like e.g. 2.9 being shown as 2. Yes it should be done for the actual log.
2. Will attempt to do.
3. It's a "just in case" thing and probably isn't really needed. I was wondering if the file couldn't be accessed if another call thread is writing to it.